sbi/neural_nets/density_estimators/categorical_net.py

```diff
@@ -107,38 +104,68 @@ def sample(self, sample_shape: torch.Size, context: Tensor) -> Tensor:
 
         # Predict Categorical ps and sample.
         ps = self.forward(context)
-        return (
-            Categorical(probs=ps)
-            .sample(sample_shape=sample_shape)
-            .reshape(sample_shape[0], -1)
-        )
+        return Categorical(probs=ps).sample(sample_shape=sample_shape)
 
 
 class CategoricalMassEstimator(DensityEstimator):
-    """Class to perform conditional density (mass) estimation
-    for a categorical RV.
+    """Conditional density (mass) estimation for a categorical RV.
+
+    The event_shape of this class is `()`.
     """
 
     def __init__(self, net: CategoricalNet) -> None:
         super().__init__(net=net, condition_shape=torch.Size([]))
         self.net = net
         self.num_categories = net.num_categories
 
-    def log_prob(self, input: Tensor, context: Tensor, **kwargs) -> Tensor:
-        return self.net.log_prob(input, context, **kwargs)
+    def log_prob(self, input: Tensor, condition: Tensor, **kwargs) -> Tensor:
+        input_iid_dim = input.shape[0]
+        input_batch_dim = input.shape[1]
+        condition_batch_dim = condition.shape[0]
+        condition_event_dims = len(condition.shape[1:])
+
+        assert condition_batch_dim == input_batch_dim, (
+            f"Batch shape of condition {condition_batch_dim} and input "
+            f"{input_batch_dim} do not match."
+        )
+
+        # Nflows needs to have a single batch dimension for condition and input.
```

Comment:
but the `CategorialEstimator` does not use `Nflows`, does it? It will be all `torch` or some future discrete flow library?

Reply:
Sorry, comment was wrong, should have been `CategoricalNet` needs a single batch dimension for condition and input.`